src/components/Search.js

```diff
@@ -17,33 +12,24 @@ class Search extends Component {
     this.setState({ searchText: e.target.value });
   }
 
-
-
-
   handleSubmit = e => {
     e.preventDefault();
     //console.log("search query value:", this.query.value)
     this.props.onSearch(this.query.value);
     //console.log("search props.onsearch:", this.props.onSearch)
    // console.log('search query value:', this.query.value);
     let query = this.query.value;
-    let path = `/search/${query}`;
-    this.props.history.push(path);
-    e.currentTarget.reset();
-
-  //  console.log('search state:', this.state.searchText);
-   // console.log("search props.onsearch:",this.props.onSearch)
-
-    
-  
+    let path = `/?q=${query}`;
```

Comment:
basically, never use `let`. You dont typically want a variable to be value `x` at one point in time, then value `y` at another. There are exceptions to this, but it violates the principle of immutability.

In this case especailly, we should be using const because it's not being reassigned.

Reply:
Ditto for all cases where using `let` but not reassigning the value.